### ISSUE TITLE
[Question]: Access token not validated?

### ISSUE THREAD

**Issue by minteo-as** [damienbod/angular-auth-oidc-client#1629]
**What Version of the library are you using?**
14

**Question**
I cannot find anything in the source code related to access_token signature validation. There is a test for id_token validation (token-validation.service.spec.ts) but nothing related to access_token. Is the access_token not validated?

**Comment by damienbod** [comment]
Only the id_token is validated, the access token is never validated in the UI, this can be any format. The API validates the access token.

**Comment by minteo-as** [comment]
Ok - is there a reason for this? I would expect, that I would need to verify that the access_token is valid and not tampered with, before allowing users to enter my guarded page.

The API that you are referring to - is this keycloak etc, or something else?

**Comment by damienbod** [comment]
An access token is never opened by the UI application. If you cannot open this, then you can validate this. There is no defined format of an access token. The API opens the access token and uses some IDP and the well known endpoints to validate the access token, for example Keycloak. The API is your backend.

**Comment by minteo-as** [comment]
In my specific case, I have a JWT access token that is signed by something like Keycloak - just as I have a signed id token in JWT format. This access token contains info, about things like the roles that my user has. If I am supposed to trust the claims in the JWT access token, and show the user some data related to his roles from the access token, then the signature must of course be valid. Otherwise the user could just alter the access token and get admin role.

It might very well be, that I am misunderstanding the role of this package. Could you please advice, what is needed for me, to ensure that the access token signature is valid. Should I maybe have an array of route guards instead of just the AutoLoginPartialRoutesGuard?

Long story short - I just want to verify the signature of any JWT token and make decisions based on verified claims.

My backend validates the access token signature and all is fine there.

**Comment by damienbod** [comment]
You have 2 or 3 tokens in the Angular app. The id_token signature is validated. In your angular application, use the id_token and only send the access token to the backend API which is a .NET application, java application etc. You need one public route to handle the callback. The backend application validates the access token including the signature.

**Comment by minteo-as** [comment]
Thanks for helping! I misunderstood the use of access_token in the frontend. Maybe too much backend coding lately..! ;-)
